engine/crates/runtime-local/src/operation_cache.rs

```diff
@@ -0,0 +1,59 @@
+use runtime::operation_cache::{OperationCache, OperationCacheFactory};
+
+pub struct InMemoryOperationCacheConfig {
+    pub limit: usize,
+}
+
+pub struct InMemoryOperationCacheFactory {
+    pub config: InMemoryOperationCacheConfig,
+}
+
+impl InMemoryOperationCacheFactory {
+    pub fn inactive() -> Self {
+        InMemoryOperationCacheFactory {
+            config: InMemoryOperationCacheConfig { limit: 0 },
+        }
+    }
+}
+
+impl Default for InMemoryOperationCacheFactory {
+    fn default() -> Self {
+        InMemoryOperationCacheFactory {
+            config: InMemoryOperationCacheConfig { limit: 1000 },
+        }
+    }
+}
+
+impl OperationCacheFactory for InMemoryOperationCacheFactory {
+    type Cache<V> = InMemoryOperationCache<V>
+    where
+        V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned;
+
+    async fn create<V>(&self) -> Self::Cache<V>
+    where
+        V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned,
+    {
+        InMemoryOperationCache {
+            inner: mini_moka::sync::Cache::builder()
+                .max_capacity(self.config.limit as u64)
+                .build(),
+        }
+    }
+}
+
+pub struct InMemoryOperationCache<V> {
+    inner: mini_moka::sync::Cache<String, V>,
+}
+
+impl<V> OperationCache<V> for InMemoryOperationCache<V>
+where
+    V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned,
+{
+    async fn insert(&self, key: String, value: V) {
+        self.inner.insert(key, value);
+    }
+
+    async fn get(&self, key: &String) -> Option<V> {
```

Comment:
:bug: **Bug**
The `get` method in the `OperationCache` implementation for `InMemoryOperationCache` should clone the value before returning it. The current implementation returns a reference to the value, which may lead to issues if the cache is modified concurrently. This can cause unexpected behavior or runtime errors.

```suggestion
    async fn get(&self, key: &String) -> Option<V> {
        self.inner.get(key).cloned()
    }
```